sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -901,18 +902,16 @@ void setAndGetSessionState(MessagingEntityType entityType) {
         setReceiver(entityType, TestUtils.USE_CASE_DEFAULT, true);
 
         StepVerifier.create(receiver.receiveMessages()
-            .take(1)
             .flatMap(message -> {
-                logger.info("SessionId: {}. LockToken: {}. LockedUntil: {}. Message received.",
-                    message.getSessionId(), message.getLockToken(), message.getLockedUntil());
-                receiver.complete(message).block(Duration.ofSeconds(15));
-                messagesPending.decrementAndGet();
-                return receiver.setSessionState(sessionState);
-            }))
-            .expectComplete()
-            .verify();
-
-        StepVerifier.create(receiver.getSessionState())
+                    logger.info("SessionId: {}. LockToken: {}. LockedUntil: {}. Message received.",
+                        message.getSessionId(), message.getLockToken(), message.getLockedUntil());
+                    assertMessageEquals(message, messageId, isSessionEnabled);
+                    messagesPending.decrementAndGet();
+                    return receiver.abandon(message)
+                        .then(receiver.setSessionState(sessionState))
+                        .then(receiver.getSessionState());
```

Comment:
What is this line for?

Reply:
This one will retrieve `session state` , so we can assert at line 917.